src/codegen/ast_interpreter.cpp

```diff
@@ -360,25 +402,66 @@ Value ASTInterpreter::executeInner(ASTInterpreter& interpreter, CFGBlock* start_
     threading::allowGLReadPreemption();
     interpreter.current_inst = NULL;
 
-    interpreter.current_block = start_block;
-    bool started = false;
-    for (auto s : start_block->body) {
-        if (!started) {
-            if (s != start_at)
-                continue;
-            started = true;
+    if (!from_start) {
+        interpreter.current_block = start_block;
+        bool started = false;
+        for (auto s : start_block->body) {
+            if (!started) {
+                if (s != start_at)
+                    continue;
+                started = true;
+            }
+
+            interpreter.current_inst = s;
+            v = interpreter.visit_stmt(s);
         }
+    } else {
+        if (should_jit)
+            interpreter.startJITing(start_block);
 
-        interpreter.current_inst = s;
-        v = interpreter.visit_stmt(s);
+        interpreter.next_block = start_block;
     }
 
     while (interpreter.next_block) {
         interpreter.current_block = interpreter.next_block;
         interpreter.next_block = 0;
 
+        if (ENABLE_BASELINEJIT && !interpreter.jit) {
+            CFGBlock* b = interpreter.current_block;
+            if (b->entry_code) {
+                should_jit = true;
+
+                try {
+                    UNAVOIDABLE_STAT_TIMER(t0, "us_timer_in_baseline_jitted_code");
+                    std::pair<CFGBlock*, Box*> rtn = b->entry_code(&interpreter, b);
```

Comment:
Ah ok this seems like a good way of handling exceptions, but it also terrifies me :)  It means that we have to make sure the code we generate corresponds to the eh_frame; I think the relevant parts of that are 1) we have to make sure the stack size matches up, and 2) we can't use any callee-save registers.  I think the rewriter already takes care of the second point, though I was hoping that we would be able to get away from that.